lib/taskana-core/src/test/java/acceptance/jobs/TaskCleanupJobAccTest.java

```diff
@@ -179,18 +179,31 @@ void should_SetNextScheduledJobBasedOnDueDateOfPredecessor_When_RunningTaskClean
     jobService.createJob(scheduledJob);
     jobsToRun = jobMapper.findJobsToRun(Instant.now());
 
-    assertThat(jobsToRun).hasSize(1);
-    assertThat(jobsToRun.get(0).getDue()).isEqualTo(firstDue);
+    assertThat(jobsToRun).extracting(ScheduledJob::getDue).containsExactly(firstDue);
 
     JobRunner runner = new JobRunner(taskanaEngine);
     runner.runJobs();
     Duration runEvery = taskanaEngineConfiguration.getCleanupJobRunEvery();
     jobsToRun = jobMapper.findJobsToRun(Instant.now().plus(runEvery));
 
-    assertThat(jobsToRun).hasSize(1);
     assertThat(jobsToRun).extracting(ScheduledJob::getDue).containsExactly(firstDue.plus(runEvery));
   }
 
+  @WithAccessId(user = "admin")
+  @Test
+  void should_ScheduleNextJobAccordingToFirstRun_When_PreviousJobNotExisting() throws Exception {
+    Instant firstRun = Instant.now().minus(2, ChronoUnit.MINUTES).truncatedTo(ChronoUnit.MILLIS);
+    Duration runEvery = Duration.ofMinutes(5);
+    taskanaEngineConfiguration.setCleanupJobRunEvery(runEvery);
+    taskanaEngineConfiguration.setCleanupJobFirstRun(firstRun);
+
+    TaskCleanupJob job = new TaskCleanupJob(taskanaEngine, null, null);
+    job.run();
+
+    List<ScheduledJob> nextJobs = getJobMapper().findJobsToRun(Instant.now().plus(runEvery));
+    assertThat(nextJobs).extracting(ScheduledJob::getDue).containsExactly(firstRun.plus(runEvery));
```

Comment:
Does it make sense to test that if the firstRun is in 5 minutes nothing will be executed?